Updates on WMS-T static temporal capabilities settings

Samweli opened the pull request:
Improvements on top of work from PR https://github.com/qgis/QGIS/pull/35741.
This PR updates the state handling and logic of changes in WMS-T static temporal capabilities, these include disabling updates on temporal capabilities if the WMS-T static temporal capabilities group in raster layer properties is not checked.

example
![updates_on_wmst_settings](https://user-images.githubusercontent.com/2663775/80321861-67849400-8829-11ea-8f17-c9f2a29153ff.gif)

elpaso commented on src/app/qgisapp.cpp:
```diff
@@ -11056,6 +11057,40 @@ void QgisApp::projectCrsChanged()
+      QgsProviderMetadata *metadata = QgsProviderRegistry::instance()->providerMetadata(
+                                        currentLayer->providerType() );
+
+      QVariantMap uri = metadata->decodeUri( currentLayer->dataProvider()->dataSourceUri() );
```
can it be const?

Samweli replied:
There are changes to be made on the uri, making it const will prevent those changes.

elpaso commented on src/gui/raster/qgsrasterlayerproperties.cpp:
```diff
@@ -1242,21 +1244,21 @@ void QgsRasterLayerProperties::apply()
+  QgsProviderMetadata *metadata = QgsProviderRegistry::instance()->providerMetadata(
+                                    mRasterLayer->providerType() );
+  QVariantMap uri = metadata->decodeUri( mRasterLayer->dataProvider()->dataSourceUri() );
+  uri["allowTemporalUpdates"] = mWmstGroup->isChecked() ? QStringLiteral( "true" ) :  QStringLiteral( "false" );
```
I'm not 100% sure but I think you could have used `bool` values directly here, give it a try.

elpaso commented:
@Samweli nice use of the metadata API :)

I've also found some missing QStringLiteral in the variant map indxes [...] would you please check them all out?

Samweli commented:
> @Samweli nice use of the metadata API :)
> 
> I've also found some missing QStringLiteral in the variant map indxes [...] would you please check them all out?

Sure @elpaso

Samweli committed:
using QStringLiteral for QVariantMap indexes

nyalldawson commented:
This looks good to me, except for the note about 579d8d16fa3fd271472c05683f56247fed0d10b5 in https://github.com/qgis/QGIS/pull/35741 (also needs to be rebased to avoid the conflicts)

timlinux commented:
Nice stuff @Samweli !

Samweli commented:
> This looks good to me, except for the note about [579d8d1](https://github.com/qgis/QGIS/commit/579d8d16fa3fd271472c05683f56247fed0d10b5) in #35741 (also needs to be rebased to avoid the conflicts)

Cool, I have updated the PR with a rebase and reverted changes from [579d8d1](https://github.com/qgis/QGIS/commit/579d8d16fa3fd271472c05683f56247fed0d10b5).

Samweli commented:
@nyalldawson @elpaso this one is also ready for merge. thank you.
